src/aem/azext_aem/custom.py

```diff
@@ -0,0 +1,546 @@
+                    pri_cfg.update({
+                        account_name + '.hour.key': disk['key'],
+                        account_name + '.minute.key': disk['key'],
+                    })
+                    pub_cfg.update({
+                        account_name + '.hour.uri': disk['table_endpoint'] + '$MetricsHourPrimaryTransactionsBlob',
+                        account_name + '.minute.uri': disk['table_endpoint'] + '$MetricsMinutePrimaryTransactionsBlob',
+                        account_name + '.hour.name': disk['account_name'],
+                        account_name + '.minute.name': disk['account_name']
+                    })
+
+        pub_cfg['wad.isenabled'] = 0
+        return pub_cfg, pri_cfg
+
+    def _get_aem_extension(self):
+        existing_ext = None
+        if self._vm.instance_view.extensions:
+            full_type_name = '.'.join([self._extension['publisher'], self._extension['name']])
+            existing_ext = next((x for x in self._vm.instance_view.extensions
+                                 if x.type and (x.type.lower() == full_type_name.lower())), None)
+        return existing_ext
+
+    def _get_disk_info(self):
+        from msrestazure.tools import parse_resource_id
+        disks_info = {}
+        disks_info['managed_disk'] = bool(getattr(self._vm.storage_profile.os_disk, 'managed_disk', None))
+        if disks_info['managed_disk']:
+            res_info = parse_resource_id(self._vm.storage_profile.os_disk.managed_disk.id)
+            disk = self._vm_client.disks.get(res_info['resource_group'], res_info['name'])
+            disks_info['os_disk'] = {
+                'name': disk.name,
+                'size': disk.disk_size_gb,
+                'is_premium': disk.sku.tier.lower() == 'premium',
+                'caching': self._vm.storage_profile.os_disk.caching.value,
+            }
+            disks_info['data_disks'] = []
+            for data_disk in self._vm.storage_profile.data_disks:
+                res_info = parse_resource_id(data_disk.managed_disk.id)
+                disk = self._vm_client.disks.get(res_info['resource_group'], res_info['name'])
+                disks_info['data_disks'].append({
+                    'name': disk.name,
+                    'size': disk.disk_size_gb,
+                    'is_premium': disk.sku.tier.lower() == 'premium',
+                    'caching': data_disk.caching.value,
+                    'lun': data_disk.lun
+                })
+        else:
+            storage_accounts = list(self._storage_client.storage_accounts.list())
+            blob_uri = self._vm.storage_profile.os_disk.vhd.uri
+            parts = list(filter(None, blob_uri.split('/')))
+            storage_account_name = parts[1].split('.')[0]
+            disk_name, container_name = parts[-1], parts[-2]
+            storage_account = [x for x in storage_accounts if x.name.lower() == storage_account_name.lower()][0]
```

Comment:
Use `next()` like you do for `existing_ext` above?
I think this would have IndexError if not storage account matched.